### ISSUE TITLE
Using settings from Proguard.html, but still getting errors

### ISSUE THREAD

**Issue by chrisvire** [awslabs/aws-sdk-android-samples#100]
I used the settings from Proguard.html that are included in the aws-android-sdk-2.3.8:

```
# Class names are needed in reflection
-keepnames class com.amazonaws.**
# Request handlers defined in request.handlers
-keep class com.amazonaws.services.**.*Handler
# The following are referenced but aren't required to run
-dontwarn com.fasterxml.jackson.**
-dontwarn org.apache.commons.logging.**
# Android 6.0 release removes support for the Apache HTTP client
-dontwarn org.apache.http.**
# The SDK has several references of Apache HTTP client
-dontwarn com.amazonaws.http.**
-dontwarn com.amazonaws.metrics.**
```

I was still getting the following output from gradle:

> Note: com.amazonaws.com.google.gson.internal.$Gson$Types$ParameterizedTypeImpl calls 'Class.getEnclosingClass'
> Warning: com.amazonaws.auth.policy.conditions.S3ConditionFactory: can't find referenced class com.amazonaws.services.s3.model.CannedAccessControlList
> Warning: com.amazonaws.auth.policy.conditions.S3ConditionFactory: can't find referenced class com.amazonaws.services.s3.model.CannedAccessControlList
> Warning: com.amazonaws.auth.policy.conditions.S3ConditionFactory: can't find referenced class com.amazonaws.services.s3.model.CannedAccessControlList
> Warning: com.amazonaws.auth.policy.conditions.S3ConditionFactory: can't find referenced class com.amazonaws.services.s3.model.CannedAccessControlList
> Note: com.amazonaws.metrics.AwsSdkMetrics: can't find dynamically referenced class com.amazonaws.metrics.internal.cloudwatch.DefaultMetricCollectorFactory
> Note: com.amazonaws.com.google.gson.internal.UnsafeAllocator accesses a declared field 'theUnsafe' dynamically
> Note: there were 1 classes trying to access enclosing classes using reflection.
>       You should consider keeping the inner classes attributes
>       (using '-keepattributes InnerClasses').
>       (http://proguard.sourceforge.net/manual/troubleshooting.html#attributes)
> Note: there were 1 unresolved dynamic references to classes or interfaces.
>       You should check if you need to specify additional program jars.
>       (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclass)
> Note: there were 1 accesses to class members by means of introspection.
>       You should consider explicitly keeping the mentioned class members
>       (using '-keep' or '-keepclassmembers').
>       (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclassmember)
> Warning: there were 4 unresolved references to classes or interfaces.
>          You may need to add missing library jars or update their versions.
>          If your code works fine without the missing classes, you can suppress
>          the warnings with '-dontwarn' options.
>          (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
> 

I did some googling and found that if I added this line to my proguard configuration file then it didn't fail to build (and the app worked):

`-dontwarn com.amazonaws.auth.policy.conditions.S3ConditionFactory`

I still get these Notes and I am not sure if I should be concerned about them:

> Note: com.amazonaws.com.google.gson.internal.$Gson$Types$ParameterizedTypeImpl calls 'Class.getEnclosingClass'
> Note: com.amazonaws.metrics.AwsSdkMetrics: can't find dynamically referenced class com.amazonaws.metrics.internal.cloudwatch.DefaultMetricCollectorFactory
> Note: com.amazonaws.com.google.gson.internal.UnsafeAllocator accesses a declared field 'theUnsafe' dynamically
> Note: there were 1 classes trying to access enclosing classes using reflection.
>       You should consider keeping the inner classes attributes
>       (using '-keepattributes InnerClasses').
>       (http://proguard.sourceforge.net/manual/troubleshooting.html#attributes)
> Note: there were 1 unresolved dynamic references to classes or interfaces.
>       You should check if you need to specify additional program jars.
>       (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclass)
> Note: there were 1 accesses to class members by means of introspection.
>       You should consider explicitly keeping the mentioned class members
>       (using '-keep' or '-keepclassmembers').
>       (http://proguard.sourceforge.net/manual/troubleshooting.html#dynamicalclassmember)
>

**Comment by minbi** [comment]
You are getting error after minifying and obfuscating?
Which samples are you using? What are the errors you are getting?

**Comment by chrisvire** [comment]
Oops.  I didn't think that anything was created for this.  I worked around it, but I should at least report my problem.  Just a sec...

**Comment by chrisvire** [comment]
So here are my questions:

1) Why did I have to add the "-dontwarn com.amazonaws.auth.policy.conditions.S3ConditionFactory" line?
2) Should I be concerned about the notes reported?

**Comment by wdane** [comment]
Could you clarify if you are using S3 in your app? If so is the version of the S3 jar the same as the core jar?

**Comment by chrisvire** [comment]
I am not using S3 in my app. I am only referencing core and mobileanalytics.  

Once difference (than how other people are using it) is that I have an AAR which references core and mobileanalytics and then my android app builds referencing the aar (and does proguard).

**Comment by chrisvire** [comment]
ping

**Comment by karthiksaligrama** [comment]
are using using a jar downloaded from https://aws.amazon.com/mobile/sdk ?  if yes, then try switching to gradle dependency and it should fix your problem. This issue should be fixed in the next release of the sdk though.

**Comment by karthiksaligrama** [comment]
fixed with 2.4.0
